promq/autocomplete/earley/suggested_types_test.go

```diff
@@ -282,3 +513,19 @@ func safeRead(sp *string) string {
 	}
 	return *sp
 }
+
+func isEqualTypes(actual []TokenType, expected []TokenType) bool {
+	if len(actual) == 0 && len(expected) == 0 {
+		return true
+	}
+	sort.Slice(actual, func(i, j int) bool {
+		return actual[i] > actual[j]
+	})
+	sort.Slice(expected, func(k, j int) bool {
+		return expected[k] > expected[j]
+	})
+	if !reflect.DeepEqual(actual, expected) {
+		return false
+	}
+	return true
+}
```

Comment:
nit: i prefer the more succinct variant:

```
func isEqualTypes(actual interface{}, expected interface{}) bool {
	return sets.NewString(actual.([]string)...).Equal(sets.NewString(expected.([]string)...))
}
```

Reply:
I like this but I cannot convert a TokenType array to string array and end up with error.